pkg/kubelet/kubelet_test.go

```diff
@@ -405,6 +407,66 @@ func TestSyncPodsStartPod(t *testing.T) {
 	fakeRuntime.AssertStartedPods([]string{string(pods[0].UID)})
 }
 
+func TestSyncPodsDeletesWhenSourcesAreReadyPerQOS(t *testing.T) {
```

Comment:
this was really helpful to aid understanding, thanks!

Reply:
Thanks for the review!